subsys/fb/cfb_shell.c

```diff
@@ -0,0 +1,355 @@
+		    width);
+
+	return err;
+}
+
+static int cmd_invert(const struct shell *shell, size_t argc, char *argv[])
+{
+	int err;
+
+	if (!dev) {
+		shell_error(shell, HELP_INIT);
+		return -ENODEV;
+	}
+
+	err = cfb_framebuffer_invert(dev);
+	if (err) {
+		shell_error(shell, "Error inverting Framebuffer");
+		return err;
+	}
+
+	cmd_cfb_print(shell, 0, 0, "");
+
+	shell_print(shell, "Framebuffer Inverted");
+
+	return err;
+}
+
+static int cmd_get_fonts(const struct shell *shell, size_t argc, char *argv[])
+{
+	int err = 0;
+	u8_t font_height;
+	u8_t font_width;
+
+	if (!dev) {
+		shell_error(shell, HELP_INIT);
+		return -ENODEV;
+	}
+
+	for (int idx = 0; idx < cfb_get_numof_fonts(dev); idx++) {
+		if (cfb_get_font_size(dev, idx, &font_width, &font_height)) {
+			break;
+		}
+		shell_print(shell, "idx=%d height=%d width=%d", idx,
+			    font_height, font_width);
+	}
+
+	return err;
+}
+
+static int cmd_get_device(const struct shell *shell, size_t argc, char *argv[])
+{
+	int err = 0;
+
+	if (!dev) {
+		shell_error(shell, HELP_INIT);
+		return -ENODEV;
+	}
+
+	shell_print(shell, "Framebuffer Device: %s", DISPLAY_DRIVER);
+
+	return err;
+}
+
+static int cmd_get_param(const struct shell *shell, size_t argc, char *argv[])
+{
+	int param = 0;
+	static const char * const param_name[] = {
+		"height", "width", "ppt", "rows", "cols"};
+	const char *param_filter = argv[1];
+
+	if (!dev) {
+		shell_error(shell, HELP_INIT);
+		return -ENODEV;
+	}
+
+	if (!strcmp(param_filter, "all")) {
+		param = -1;
+	} else if (!strcmp(param_filter, param_name[0])) {
+		param = CFB_DISPLAY_HEIGH;
+	} else if (!strcmp(param_filter, param_name[1])) {
+		param = CFB_DISPLAY_WIDTH;
+	} else if (!strcmp(param_filter, param_name[2])) {
+		param = CFB_DISPLAY_PPT;
+	} else if (!strcmp(param_filter, param_name[3])) {
+		param = CFB_DISPLAY_ROWS;
+	} else if (!strcmp(param_filter, param_name[4])) {
+		param = CFB_DISPLAY_COLS;
+	} else {
+		shell_error(shell, "wrong parameter name");
+		shell_help(shell);
+		return 1;
+	}
+
+	if (param >= CFB_DISPLAY_HEIGH) {
+		shell_print(shell, "param: %s=%d", param_name[param],
+		      cfb_get_display_parameter(dev, param));
+	} else {
+		for (unsigned int i = 0; i <= CFB_DISPLAY_COLS; i++) {
+			shell_print(shell, "param: %s=%d", param_name[i],
+			      cfb_get_display_parameter(dev, i));
+
+		}
+	}
+
+	return 0;
+}
+
+static int cmd_init(const struct shell *shell, size_t argc, char *argv[])
+{
+	int err;
+
+	dev = device_get_binding(DISPLAY_DRIVER);
+
+	if (dev == NULL) {
+		shell_error(shell, "Device not found");
+		return -ENODEV;
+	}
+
+	err = cfb_framebuffer_init(dev);
+	if (err) {
+		shell_error(shell, "Framebuffer initialization failed!");
+		return err;
+	}
+
+	shell_print(shell, "Framebuffer initialized: %s", DISPLAY_DRIVER);
+
+	cmd_clear(shell, argc, argv);
+
+	return err;
+}
+
+SHELL_CREATE_STATIC_SUBCMD_SET(cfb_cmds) {
+	SHELL_CMD(init, NULL, HELP_NONE, cmd_init),
```

Comment:
This could be used with `_ARG`, shell will check arguments count.

`SHELL_CMD_ARG(init, NULL, HELP_NONE, cmd_init, 1, 0),`